src/System.Memory/ref/System.Memory.cs

```diff
@@ -112,6 +112,24 @@ public static partial class MemoryExtensions
 }
 namespace System.Buffers
 {
+    public sealed partial class ArrayBufferWriter : System.Buffers.IBufferWriter<byte>, System.IDisposable
+    {
+        public ArrayBufferWriter() { }
+        public ArrayBufferWriter(int initialCapacity) { }
+        public int BytesAvailable { get { throw null; } }
+        public int BytesWritten { get { throw null; } }
+        public int Capacity { get { throw null; } }
+        public System.ReadOnlyMemory<byte> OutputAsMemory { get { throw null; } }
+        public System.ReadOnlySpan<byte> OutputAsSpan { get { throw null; } }
+        public long TotalBytesWritten { get { throw null; } }
+        public void CopyToAndReset(System.IO.Stream stream) { }
+        public System.Threading.Tasks.Task CopyToAndResetAsync(System.IO.Stream stream, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public void Dispose() { }
+        public void Reset() { }
```

Comment:
Does Reset go back to 0 or initialCapacity? Do we need done that takes a capacity?

Reply:
Also, Reset seems a little confusing if it doesn't reset everything, but presumably it doesn't reset TotalBytesWritten? Should that property be removed? Can't consumers track that themselves?

Reply:
If we are removing the CopyTo APIs, we should remove the property `TotalBytesWritten` as well.

Reply:
Yes, I think they both should be removed.